core/trino-parser/src/main/java/io/trino/sql/parser/SqlParser.java

```diff
@@ -65,7 +65,6 @@ public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int
             .specialRule(SqlBaseParser.RULE_query, "<query>")
             .specialRule(SqlBaseParser.RULE_type, "<type>")
             .specialToken(SqlBaseLexer.INTEGER_VALUE, "<integer>")
-            .ignoredRule(SqlBaseParser.RULE_nonReserved)
```

Comment:
Do you think we can remove `ignoredRule` from `ErrorHandler` now?
AFAIK, `nonReserved` was the only ignored rule.